### PR TITLE
net-im/spectrum2: use correct python eclass

### PR DESCRIPTION
**Pull request by ConiKost**
Package-Manager: Portage-2.3.69, Repoman-2.3.16
Signed-off-by: Conrad Kostecki <conrad@kostecki.com>

### CODE REVIEW

File: net-im/spectrum2/spectrum2-2.0.10.1-r2.ebuild [hunk 1]
```diff
@@ -18,6 +19,7 @@ IUSE="doc frotz irc mysql postgres purple sms +sqlite test twitter whatsapp xmpp
 REQUIRED_USE="|| ( mysql postgres sqlite )"
 
 RDEPEND="
+	test? ( ${PYTHON_DEPS} )
```
Comment (a17r) [inline net-im/spectrum2/spectrum2-2.0.10.1-r2.ebuild]: test deps are in `RDEPEND` all of a sudden?

Comment (ConiKost) [reply net-im/spectrum2/spectrum2-2.0.10.1-r2.ebuild]: Well, CI warnings says, I have to set PYTHON_DEPS in RDEPEND? Is this wrong in this case?

File: net-im/spectrum2/spectrum2-2.0.10.1-r2.ebuild [hunk 2]
```diff
@@ -10,6 +10,7 @@ inherit cmake-utils python-single-r1 systemd user
 DESCRIPTION="An open source instant messaging transport"
 HOMEPAGE="https://www.spectrum.im"
 SRC_URI="https://github.com/SpectrumIM/spectrum2/archive/${PV}.tar.gz -> ${P}.tar.gz"
+REQUIRED_USE="test? ( ${PYTHON_DEPS} )"
```
[the remark this answers is not in the document]
Comment (ConiKost) [reply net-im/spectrum2/spectrum2-2.0.10.1-r2.ebuild]: Sorry, copy/paste error, fixed.

File: net-im/spectrum2/spectrum2-2.0.10.1-r2.ebuild (hunk 1 above)
Comment (a17r) [reply net-im/spectrum2/spectrum2-2.0.10.1-r2.ebuild]: If python is only used during build-time, then this ebuild should use `python-any-r1` instead. But if this ebuild is installing python files used at runtime, then I doubt that it is correct as conditional on `USE=test`?

Comment (ConiKost) [reply net-im/spectrum2/spectrum2-2.0.10.1-r2.ebuild]: I see, than I used it indeed wrong. I changed not to ```python-any-r1``` and moved PYTHON_DEPS back to ```DEPEND```. Python is only used for tests. Is this correct this way?

**Comment by andrey-utkin** [comment]
Hi @ConiKost!
Thanks for your work.
Could you please specify the warnings you are dealing with, in commit message?

File: net-im/spectrum2/spectrum2-2.0.10.1-r2.ebuild (hunk 1 above)
Comment (mgorny) [reply net-im/spectrum2/spectrum2-2.0.10.1-r2.ebuild]: Yes but python-any-r1 doesn't have PYTHON_REQUIRED_USE.

**Comment by a17r** [comment]
> Could you please specify the warnings you are dealing with, in commit message?

git summary should describe the switch of eclass rather than the cause of action anyway.

File: net-im/spectrum2/spectrum2-2.0.10.1-r2.ebuild (hunk 1 above)
Comment (ConiKost) [reply net-im/spectrum2/spectrum2-2.0.10.1-r2.ebuild]: Thanks, removed.

**Comment by ConiKost** [comment]
> git message should describe the switch of eclass rather than the cause of action anyway.

Done.

**Comment by ConiKost** [comment]
Sorry, should be now ok, repoman is locally now happy.

File: net-im/spectrum2/spectrum2-2.0.10.1-r2.ebuild [hunk 3]
```diff
@@ -49,11 +49,11 @@ RDEPEND="
 	whatsapp? ( net-im/transwhat )"
 
 DEPEND="
-	${PYTHON_DEPS}
 	${RDEPEND}
 	doc? ( app-doc/doxygen )
 	test? (
-		dev-python/sleekxmpp[${PYTHON_USEDEP}]
+		${PYTHON_DEPS}
+		dev-python/sleekxmpp
```
Comment (a17r) [inline net-im/spectrum2/spectrum2-2.0.10.1-r2.ebuild]: why did you drop `${PYTHON_USEDEP}` here?

Comment (ConiKost) [reply net-im/spectrum2/spectrum2-2.0.10.1-r2.ebuild]: Because that was the cause of the repoman failure?

Comment (a17r) [reply net-im/spectrum2/spectrum2-2.0.10.1-r2.ebuild]: devmanual's got you covered: https://wiki.gentoo.org/wiki/Project:Python/python-any-r1#Package_with_additional_dependencies

Comment (ConiKost) [reply net-im/spectrum2/spectrum2-2.0.10.1-r2.ebuild]: Ok, please have a look, if this is now correct.

**Pull request closed without merging**